### PR TITLE
829279- exporting data grid

### PR DESCRIPTION
**Pull request by NirmalKumarYuvaraj**
I have added UG documentation for MAUI DataGrid Exporting feature (Export to PDF).

### CODE REVIEW

File: MAUI/DataGrid/export-to-pdf.md [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+### RowExporting
+
+The `DataGridRowPdfExportingEventHandler` delegate allows customizing the styles of record rows and group caption rows. The `RowExporting` event is triggered with `DataGridRowPdfExportingEventArgs` that contains the following properties:
+
+* `PdfGrid`: Customizes the pdfGrid properties such as `Background`, `CellPadding`, `CellSpacing`, and so on.
+* `PdfRow`: Specifies the `PDFGridRow` to be exported. Customizes the properties of a particular row. 
+* `Record`: Gets the collection of exported underlying data objects.
+* `RowType`: Specifies the row type using `ExportRowType` enum. Checks the row type and applies different styles based on the row type.
+
+You can use this event to customize the properties of the grid rows exported to PDF. The following code example illustrates how to change the background color of the record rows and caption summary rows when exporting.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/export-to-pdf.md:584]: Grammar check and revised content:

The `DataGridRowPdfExportingEventHandler` delegate allows for customizing the styles of record rows and group caption rows. The `RowExporting` event is triggered with `DataGridRowPdfExportingEventArgs`, which contains the following properties:

* `PdfGrid`: Customizes the properties of the `pdfGrid`, such as `Background`, `CellPadding`, `CellSpacing`, and more.
* `PdfRow`: Specifies the `PDFGridRow` to be exported and allows customization of the properties of a particular row.
* `Record`: Retrieves the collection of exported underlying data objects.
* `RowType`: Specifies the row type using the `ExportRowType` enum. It checks the row type and applies different styles based on the type.

You can use this event to customize the properties of the grid rows that are exported to PDF. The following code example demonstrates how to change the background color of the record rows and caption summary rows during the export process.

File: MAUI/DataGrid/export-to-pdf.md [hunk 2]
```diff
@@ -0,0 +1,665 @@
...
+
+* `ColumnName`: Specifies the column name (MappingName) of the exporting cell. Applies formatting for a particular column by checking the `ColumnName`.
+
+* `Handled`: Determines whether the cell is exported to PDF or not.
+
+* `PdfGrid`: Specifies the `PDFGridCell` to be exported. Customizes Background, Foreground, Font, Alignment, etc., properties of a particular cell.
+
+* `Record`: Gets the collection of exported underlying data objects.
+
+You can use this event to customize the properties of the grid cells exported to PDF. The following code example illustrates how to customize the background color, foreground color, and cell value of the header cells, record cells, and caption summary cells when exporting.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/export-to-pdf.md:621]: The `DataGridCellPdfExportingEventHandler` delegate allows for customizing the styles of header cells, record cells, and group caption cells. The `CellExporting` event is triggered with `DataGridCellPdfExportingEventArgs`, which contains the following properties:

* `CellType`: Specifies the cell type using the `ExportCellType` enum. It checks the cell type and applies different cell styles based on the type.

* `CellValue`: Contains the actual exported value used to format the PDF using the `Range` property.

* `ColumnName`: Specifies the column name (MappingName) of the exporting cell. It applies formatting for a particular column by checking the `ColumnName`.

* `Handled`: Determines whether the cell is exported to PDF or not.

* `PdfGrid`: Specifies the `PDFGridCell` to be exported. It customizes properties such as Background, Foreground, Font, Alignment, etc., for a particular cell.

* `Record`: Retrieves the collection of underlying data objects being exported.

You can use this event to customize the properties of grid cells that are exported to PDF. The code example below demonstrates how to customize the background color, foreground color, and cell value of header cells, record cells, and caption summary cells during the export process.

File: MAUI/DataGrid/export-to-pdf.md [hunk 3]
```diff
@@ -0,0 +1,665 @@
...
+description: Learn here all about PDF Exporting support in Syncfusion MAUI DataGrid (SfDataGrid) control and more about it.
+platform: MAUI
+control: SfDataGrid
+documentation: UG
+---
+# Export To PDF in MAUI DataGrid (SfDataGrid)
+
+The SfDataGrid provides extensive support for exporting data to PDF, offering various customization options to meet your specific requirements. With this feature, you can customize the appearance of the exported PDF, exclude specific columns or headers, and even define custom row heights and column widths and so on.
+
+If you are using nuget package in the package, the following NuGet package should be installed to export the SfDataGrid to PDF file.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:13]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 4]
```diff
@@ -0,0 +1,665 @@
...
+<th> Project </th>
+<th> Required package </th>
+</tr>
+<tr>
+<td> .NET MAUI </td>
+<td> Syncfusion.Maui.DataGridExport</td>
+</tr>
+</table>
+
+You can export the SfDataGrid to PDF by using the following extension methods present in the Syncfusion.Maui.DataGrid.Exporting namespace.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:26]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 5]
```diff
@@ -0,0 +1,665 @@
...
+                    ItemsSource="{Binding OrderInfoCollection}" >
+        <syncfusion:SfDataGrid.Columns>
+            <syncfusion:DataGridNumericColumn MappingName="OrderID" HeaderText="Order ID" Format="d"/>
+            <syncfusion:DataGridTextColumn MappingName="CustomerID" HeaderText="Customer ID"/>
+            <syncfusion:DataGridTextColumn MappingName="Customer" HeaderText="Customer"/>
+            <syncfusion:DataGridTextColumn MappingName="ShipCountry" HeaderText="Ship Country"/>
+            <syncfusion:DataGridTextColumn MappingName="ShipCity" HeaderText="Ship City"/>
+        </syncfusion:SfDataGrid.Columns>
+    </syncfusion:SfDataGrid>
+</StackLayout>
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:50]: Aligned the code properly.

File: MAUI/DataGrid/export-to-pdf.md [hunk 6]
```diff
@@ -0,0 +1,665 @@
...
+{% endhighlight %}
+{% endtabs %}
+
+## ExportToPdf
+
+You can export the data to PDF by using the `DataGridPdfExportingController.ExportToPdf` method by passing the SfDataGrid as an argument.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:56]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 7]
```diff
@@ -0,0 +1,665 @@
...
+    SaveService saveService = new();
+    saveService.SaveAndView("ExportFeature.pdf", "application/pdf", stream);
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+## ExportToPdfGrid
+
+You can also export the data to PDF by using the `DataGridPdfExportingController.ExportToPdfGrid` method by passing the SfDataGrid as an argument.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:78]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 8]
```diff
@@ -0,0 +1,665 @@
...
+    pdfDoc.Close(true);
+    SaveService saveService = new();
+    saveService.SaveAndView("ExportFeature.pdf", "application/pdf", stream);
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to Excel format" src="Images\export-to-pdf\maui-datagrid-export-datagrid-to-pdf.png" width="689"/>
+
+N> SfDataGrid cannot export the DataGridTemplateColumn to PDF or Excel,  since we cannot get the loaded views and draw them with the particular range, values etc from DataGridTemplateColumn.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:104]: updated content

File: MAUI/DataGrid/export-to-pdf.md [hunk 9]
```diff
@@ -0,0 +1,665 @@
...
+
+## Exporting options
+
+### Exclude columns when exporting
+
+By default, all the columns (including hidden columns) in the SfDataGrid will be exported to PDF. To exclude some particular columns when exporting to PDF, add those columns to `DataGridPdfExportingOption.ExcludeColumns` list.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:110]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 10]
```diff
@@ -0,0 +1,665 @@
...
+option.ExcludedColumns = list;
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format with certain columns excluded" src="Images\export-to-pdf\maui-datagrid-exclude-columns.png" width="689"/>
+
+
+### Getting PDF document 
+
+The `DataGridPdfExportingOption.PdfDocument` allows exporting the SfDataGrid to an existing or a new PDF document.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:127]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 11]
```diff
@@ -0,0 +1,665 @@
...
+list.Add("OrderID");
+list.Add("CustomerID");
+option.ExcludedColumns = list;
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format with certain columns excluded" src="Images\export-to-pdf\maui-datagrid-exclude-columns.png" width="689"/>
+
+
+### Getting PDF document
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:125]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 12]
```diff
@@ -0,0 +1,665 @@
...
+pdfDocument.Pages.Add();
+pdfDocument.Pages.Add();
+option.StartPageIndex = 1;
+option.PdfDocument = pdfDocument;
+{% endhighlight %}
+{% endtabs %}
+
+### Getting Columns for customization
+
+Using the property `Columns` you can get or set the `System.Collections.IEnumerable` columns collection which contains all the columns that are to be exported. The columns in the ExcludedColumns List will not be a member of the Columns collection.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:143]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 13]
```diff
@@ -0,0 +1,665 @@
...
+
+### Column header on each page
+
+You can show or hide the column headers on each page of the exported PDF document by using the `DataGridPdfExportingOption.CanRepeatHeaders` property. The default value is true.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:147]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 14]
```diff
@@ -0,0 +1,665 @@
...
+DataGridPdfExportingOption option = new DataGridPdfExportingOption();
+option.CanExportGroupSummary = true;
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format with group summaries" src="Images\export-to-pdf\maui-datagrid-export-group-summary.png" width="689"/>
+
+#### Exporting the selected rows of SfDataGrid
+
+SfDataGrid allows you to export only the currently selected rows in the grid to the document using the `DataGridPdfExportingController.ExportToPdf` method by passing the instance of the SfDataGrid and `SfDataGrid.SelectedRows` collection as an argument.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:214]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 15]
```diff
@@ -0,0 +1,665 @@
...
+
+Refer the below code to export the selected rows alone to the PDF document.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:216]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 16]
```diff
@@ -0,0 +1,665 @@
...
+{% tabs %}
+{% highlight c# %}
+ObservableCollection<object> selectedItems = dataGrid.SelectedRows;
+var pdfDoc = pdfExport.ExportToPdf(this.dataGrid, selectedItems);
+{% endhighlight %}
+{% endtabs %}
+
+### Export all columns in one page
+
+Gets or sets a value indicating whether all the columns should be fitted on a page or not. To export all the columns in one page, set the`DataGridPdfExportingOption.CanFitAllColumnsInOnePage` property to `true`.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:227]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 17]
```diff
@@ -0,0 +1,665 @@
...
+option.CanFitAllColumnsInOnePage = true;
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format with all columns fit in view" src="Images\export-to-pdf\maui-datagrid-fit-all-columns-in-one-page.png" width="689"/>
+
+
+### Exporting the grid from a specified page and point
+
+The SfDataGrid allows exporting the data to a particular staring position on a particular PDF page using the following options:
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:241]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 18]
```diff
@@ -0,0 +1,665 @@
...
+
+* StartPageIndex
+* StartPoint 
+
+#### StartPageIndex 
+
+The SfDataGrid allows exporting the data to a particular page in the PDF document by using the `DataGridPdfExportingOption.StartPageIndex` property.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:248]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 19]
```diff
@@ -0,0 +1,665 @@
...
+{% tabs %}
+{% highlight c# %}
+DataGridPdfExportingOption option = new DataGridPdfExportingOption();
+option.StartPageIndex = 2;
+{% endhighlight %}
+{% endtabs %}
+
+#### StartPoint
+
+The SfDataGrid allows exporting the data to a particular x,y starting point in the PDF page by using the `DataGridPdfExportingOption.StartPoint` property.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:259]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 20]
```diff
@@ -0,0 +1,665 @@
...
+{% highlight c# %}
+DataGridPdfExportingOption option = new DataGridPdfExportingOption();
+option.StartPoint = new Syncfusion.Drawing.PointF(0, 500);
+{% endhighlight %}
+{% endtabs %}
+
+
+### Applying styles while exporting
+
+The SfDataGrid allows exporting the data with the applied DefaultStyle by setting the `DataGridPdfExportingOption.CanApplyGridStyle` property to `true`. By default, data will be exported without the DefaultStyle.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:271]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 21]
```diff
@@ -0,0 +1,665 @@
...
+{% tabs %}
+{% highlight c# %}
+DataGridPdfExportingOption option = new DataGridPdfExportingOption();
+option.CanApplyGridStyle = true;
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format with style applied" src="Images\export-to-pdf\maui-datagrid-export-with-defaultstyle.png" width="689"/>
+
+You can also customize the following styles while exporting to PDF:
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:282]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 22]
```diff
@@ -0,0 +1,665 @@
...
+
+* BottomTableSummaryStyle 
+* GroupCaptionStyle
+* HeaderStyle
+* RecordStyle 
+* TopTableSummaryStyle
+* GroupSummaryStyle
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:289]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 23]
```diff
@@ -0,0 +1,665 @@
...
+    StringFormat = new PdfStringFormat() { Alignment = PdfTextAlignment.Right, CharacterSpacing = 3f, WordSpacing = 10f }
+};
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format with style applied for table summary rows at the Top" src="Images\export-to-pdf\maui-datagrid-group-summary-style.png" width="689"/>
+
+### Customizing borders
+
+The SfDataGrid allows customizing the grid borders as follows using the `DataGridPdfExportingOption.GridLineType` property:
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:418]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 24]
```diff
@@ -0,0 +1,665 @@
...
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format without border lines" src="Images\export-to-pdf\maui-datagrid-gridlinetype-none.png" width="689"/>
+
+## Setting header and footer
+
+The SfDataGrid provides a way to display additional content at the top (header) or bottom (footer) of the page when exporting to PDF by handling the `DataGridPdfExportingController.HeaderAndFooterExporting` event.
+
+You can insert string in the header and footer in PdfHeaderFooterEventHandler. Setting `PdfPageTemplateElement` as `PdfHeaderFooterEventArgs.PdfDocumentTemplate.Top` loads the content at the top of the page. Setting the `PdfPageTemplateElement` as `PdfHeaderFooterEventArgs.PdfDocumentTemplate.Bottom` loads the content at the bottom of the page.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:481]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 25]
```diff
@@ -0,0 +1,665 @@
...
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format with header and footer" src="Images\export-to-pdf\maui-datagrid-header-footer-event.png" width="689"/>
+
+## Change PDF page orientation
+
+You can change the page orientation of the PDF document when exporting. Default page orientation is portrait.
+
+To change the page orientation, export PDF grid value by using the ExportToPdfGrid method. Then, draw that PDF grid into a PDF document by changing the `PageSettings.Orientation` property of PDF document.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:508]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 26]
```diff
@@ -0,0 +1,665 @@
...
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Export DataGrid to PDF format in Landscape orientation" src="Images\export-to-pdf\maui-datagrid-orientation.png" width="689"/>
+
+## Row height and column width customization
+
+### ExportColumnWidth
+
+By default, the data grid columns will be exported to PDF with `DataGridPdfExportingOption.DefaultColumnWidth` value. To export the data grid to PDF with exact column widths, set the `DataGridPdfExportingOption.CanExportColumnWidth` to `true`.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:524]: updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 27]
```diff
@@ -0,0 +1,665 @@
...
+option.CanExportRowHeight = true;
+{% endhighlight %}
+{% endtabs %}
+
+## Events
+
+The SfDataGrid provides you the following events for exporting:
+
+* `RowExporting`: Raised when exporting a row at the execution time.
+* `CellExporting`: Raised when exporting a cell at the execution time.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:573]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md (hunk 1 above)
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:584]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md (hunk 2 above)
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:621]: Updated the content

File: MAUI/DataGrid/export-to-pdf.md [hunk 28]
```diff
@@ -0,0 +1,665 @@
...
+        e.PdfGridCell.Style.BackgroundBrush = PdfBrushes.LightBlue;
+        e.PdfGridCell.Style.TextBrush = PdfBrushes.Black;
+    }        
+}
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Customize the cells in a DataGrid while exporting using the CellExporting event" src="Images\export-to-pdf\maui-datagrid-cell-exporting.png" width="689"/>
+
+## Cell customization in PDF while exporting
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:646]: Updated the topic name

File: MAUI/DataGrid/export-to-pdf.md [hunk 29]
```diff
@@ -0,0 +1,665 @@
...
+
+### Customize the cells based on column name
+
+You can customize the column style based on the row data when exporting to PDF by handling the `CellExporting` event.
```
[the remark this answers is not in the document]
Comment (NirmalKumarYuvaraj) [reply MAUI/DataGrid/export-to-pdf.md:650]: Updated the content

**Review by FarjanaParveen — APPROVED** [review]
This is fine

**Pull request merged**